pkg/events/derive/derive.go

```diff
@@ -41,12 +43,15 @@ func (t Table) Register(deriveFrom, deriveTo events.ID, deriveCondition func() b
 }
 
 // DeriveEvent takes a trace.Event and checks if it can derive additional events from it as defined by a derivationTable.
-func (t Table) DeriveEvent(event trace.Event) ([]trace.Event, []error) {
+func (t Table) DeriveEvent(event trace.Event, origArgs []trace.Argument) ([]trace.Event, []error) {
 	derivatives := []trace.Event{}
 	errors := []error{}
 	deriveFns := t[events.ID(event.EventID)]
 	for id, deriveFn := range deriveFns {
 		if deriveFn.Enabled() {
+			// at each derivation, we need use a copy of the original arguments,
+			// since they might be modified by a previous derivation.
+			event.Args = slices.Clone(origArgs)
```

Comment:
I see that you also clone the arguments slice before calling `DeriveEvent`, but if we also clone here, won't it be enough to just save the original `event.Args` before this for loop and remove the need to clone the args before calling the `DeriveEvent` function?

Reply:
I suppose that's not enough. The original event has its args cloned and is sent through the pipeline, and up to that point it's safe for the original event, since we won't be touching the original args reference in the derivation.

The same should happen in each derivation, to avoid one derivation can mess with the args reference of another.

Reply:
Let's assume there are no derivations at all. So we just clone the args and nothing happens, but we have some performance penatly.
Now if there is a derived event, then the for loop we have here will be entered and we clone the arguments - again, no harm is done.
So I still don't understand why we need to have the extra copy before calling `DeriveEvent`

Reply:
We've discussed this point already and decided to keep it as it's to ensure that the original event won't have its args messed by any derived one.